src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/GpNameAndAddress.java

```diff
@@ -0,0 +1,61 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import org.springframework.util.StringUtils;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.Split;
+
+/**
+ * Example NAD+GP+2750922,295:900'
+ */
+@Getter
+@Setter
+@Builder
+@AllArgsConstructor
+public class GpNameAndAddress extends Segment {
+
+    public static final String KEY = "NAD";
+    public static final String QUALIFIER = "GP";
+    public static final String KEY_QUALIFIER = KEY + "+" + QUALIFIER;
+    private @NonNull String identifier;
+    private @NonNull String code;
+
+    public static GpNameAndAddress fromString(String edifactString) {
+        if (!edifactString.startsWith(GpNameAndAddress.KEY_QUALIFIER)) {
+            throw new IllegalArgumentException("Can't create " + GpNameAndAddress.class.getSimpleName() + " from " + edifactString);
+        }
+        String[] keySplit = Split.byPlus(edifactString);
+        String identifier = Split.byColon(keySplit[2])[0];
+        String code = Split.byColon(keySplit[2])[1];
+        return new GpNameAndAddress(identifier, code);
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        return QUALIFIER + "+" + identifier + ":" + code;
+    }
+
+    @Override
+    protected void validateStateful() {
+        //NOP
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        if (StringUtils.isEmpty(identifier)) {
```

Comment:
Spring's `StringUtils.isEmpty()` is deprecated. In other places in your PR (e.g. FreeText.java) we're using Apache commons `StringUtils.isBlank()` - which i'd recommend as an alternative as it checks if a String is whitespace, empty ("") or null, rather than just empty ("") or null.

There may be other classes in your PR using this deprecated method.

Reply:
Replaced all instances of Spring's `StringUtils.isEmpty()` with Apache Commons `StringUtils.isBlank()`